ColorBot/ColorBot.js

```diff
@@ -118,18 +118,17 @@ bot.on("message", async message =>
                                                 });
 
                                                 /*
-                                                At this point, I'm attempting to array role through bot.colors to define what
-                                                needs to be added to colors.json [role & hex]
+                                                I have this working, and array wasn't needed because i was not defining
+                                                any form of title.
                                                 */
-                                                bot.colors[role] = 
+                                                bot.colors = 
                                                 {
                                                     role: args[3].toUpperCase(),
                                                     hex: args[2]
                                                 }
                                                 
-                                                //I think the problem is here. Once it tries to write the info, it restarts
-                                                //I have no idea why - and it shows no error, but the object does not write
-                                                fs.writeFileSync("colors.json", JSON.stringify(bot.colors, null, 4), err =>
+                                                //bot.colors does not write properly, as explained above
+                                                fs.appendFile("./colors.json", JSON.stringify(bot.colors, null, 4), err =>

```

Comment:
appendFile is an asynchronous method. You want to try appendFileSync instead https://nodejs.org/api/fs.html#fs_fs_appendfilesync_file_data_options

Reply:
or cause you're using `async` you can probably just `await` this call.

Reply:
Right, I attempted using appendFileSync, and I ended up with the same issue. I've provided a screenshot of what the result is (same as before). The only more or less "valid" idea I have is to completely rewrite the file every time.
SCREENSHOT: https://imgur.com/a/Jq652

Reply:
Writing in the wrong spot means there is some issue with how the previous stream of data was read in and the order of events for each read/write operation. Some things to consider is if you are reading/writing asynchronously are you resolving those calls?